### DISCUSSION TITLE
Where do you set isolateGlobalState when using MobX in Angular?

### DISCUSSION THREAD

**Discussion by TomMakes** [mobxjs/mobx#3385] [Q&A]
I believe the only way I can solve the error _[mobx] There are multiple, different versions of MobX active. Make sure MobX is loaded only once or use `configure({ isolateGlobalState: true })`_ in my case is by setting isolateGlobalState to true, however I can not figure out where in my Angular project do I need to set it so this error doesn't appear.

I have tried setting it in polyfills.ts, main.ts and app-routing.ts using the code:
```
import { configure } from 'mobx';
configure( { isolateGlobalState: true });
```
None of these locations worked.

**Why I need to do this:**
In my Angular application, I am using [Angular Tree Component](https://www.npmjs.com/package/@circlon/angular-tree-component) and [Redoc](https://www.npmjs.com/package/redoc), which both use MobX.

Angular Tree Component uses MobX as a direct dependency, while Redoc uses it as a peer dependency, creating two instances of MobX in my project.

**Comment by mweststrate** [comment]
> Angular Tree Component uses MobX as a direct dependency

This typically is a bug, an ideally should be fixed at that level. things like `configure({ isolateGlobalState: true })` are bandaids around that, that probably never should have been released in the first place😅

**Reply by TomMakes** [reply]
If I still want to use the bandaid, where would I apply it?

**Reply by mweststrate** [reply]
As early as possible in your scripts. however bundling is unpredictable in JS, and I don't know if Angular does anything specific here, but basically you want to import mobx first, call configure, then import all the rest in your entry point / before your app is started.
